packages/kbn-alerts-ui-shared/src/common/hooks/use_load_connector_types.ts

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { useQuery } from '@tanstack/react-query';
+import type { HttpStart } from '@kbn/core-http-browser';
+import { fetchConnectorTypes } from '../apis';
+
+export interface UseLoadActionTypesProps {
+  http: HttpStart;
+  includeSystemActions?: boolean;
+}
+
+export const useLoadActionTypes = (props: UseLoadActionTypesProps) => {
+  const { http, includeSystemActions } = props;
+
+  const queryFn = () => {
+    return fetchConnectorTypes({ http, includeSystemActions });
+  };
+
+  const { data, isLoading, isFetching, isInitialLoading } = useQuery({
```

Comment:
In our last sync, @umbopepato suggested returning the whole value of the `useQuery` hook instead of choosing specific attributes to return.  There are a lot of benefits like consistency in types, automatic type inference from the hook, chain query objects, and the ability to select which loading state you would like to check. For example, when the `enable` property is `false` the `isLoading` is `true` but the `isFetching` is `false`. What do you think? 

Same for `packages/kbn-alerts-ui-shared/src/common/hooks/use_load_connectors.ts` and `packages/kbn-alerts-ui-shared/src/common/hooks/use_load_rule_type_aad_template_fields.ts`.

Reply:
Yea I dont really mind, i think we had a bug where if we just check `isLoading`, it doesn't tell us 100% if it was loading, so we had to check `isLoading || isFetching`.